packages/transformers/webextension/src/WebExtensionTransformer.js

```diff
@@ -88,8 +92,10 @@ async function collectDependencies(
       for (const k of ['css', 'js']) {
         const assets = sc[k] || [];
         for (let j = 0; j < assets.length; ++j) {
+          if (k == 'js') {
+            asset.invalidateOnFileChange(path.join(assetDir, assets[j]));
```

Comment:
Why do we need to invalidate the manifest when the JS file changes (but not CSS?)? We also add a dependency below, so that should already capture the JS change...

Reply:
Actually this is specifically to handle the case that this JS file is updated to reference a CSS bundle, because that doesn't trigger the `WebExtensionPackager` again so the CSS doesn't get added to the manifest.

Reply:
Actually this wouldn't work for nested dependencies...any solutions?

Reply:
Ah hmm. That's more of a problem with caching in packagers. At the moment if you access other bundles from within a packager it won't be invalidated. Let me think about that.

Reply:
How is CSS importing support in JS when the JS is referenced from HTML then? We should be able to replicate the strategy here.

Reply:
The HTML packager only needs to list directly referenced bundles, and their siblings. `getReferencedBundles` handles this. It's possible that it works because the JS and CSS get placed in the same bundle group (i.e. loaded together) as the HTML file. That's due to `priority: 'parallel'` on the dependency. You could potentially also try that here.

However, that still wouldn't work for nested dependencies, i.e. async `import()` in the JS. I guess those would also need to be listed in `web_accessible_resources`?

Reply:
Yes, I'm looking into solutions for that right now. However there's another issue: since CSS is parsed outside the extension environment for content scripts, [relative paths don't work](https://stackoverflow.com/a/8864212/10930232). We need to prepend `chrome-extension://__MSG_@@extension_id__/` to any bundle referenced from a content-script CSS bundle. Do we have a way to do that?

Reply:
Yeah, if you're comfortable, we could probably just merge this as is for now, and address the caching issue separately. Maybe having something is better than being perfect? Should be a somewhat rare edge case when adding/removing a bundle.

Reply:
Yeah that's probably for the best, that edge case seems a bit difficult to tackle right now, and the MV3 support from this PR is, as far as I can tell, already way cleaner than what's available in any other bundler.